pkg/machines/components/vm/vmMigrateDialog.jsx

```diff
@@ -0,0 +1,194 @@
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+import {
+    Button,
+    Checkbox,
+    Form,
+    FormGroup,
+    Modal,
+    Radio,
+    TextInput,
+    Tooltip
+} from '@patternfly/react-core';
+
+import { migrateToUri } from '../../libvirt-dbus.js';
+import { isEmpty, isObjectEmpty } from '../../helpers.js';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+
+const _ = cockpit.gettext;
+
+const DestUriRow = ({ validationFailed, destUri, setDestUri }) => {
+    return (
+        <FormGroup label={_("Destination URI")} fieldId="dest-uri"
+                   id="dest-uri"
+                   validated={validationFailed.destUri ? "error" : "default"}>
+            <TextInput id='dest-uri-input'
+                       validated={validationFailed.destUri ? "error" : "default"}
+                       value={destUri}
+                       placeholder={cockpit.format(_("Example, $0"), "qemu+ssh://192.0.2.16/system")}
+                       onChange={setDestUri} />
+        </FormGroup>
+    );
+};
+
+const TypeRow = ({ type, setType, setUndefineSource, setStorage }) => {
+    return (
+        <FormGroup label={_("Migrated VM state")}
+                   fieldId="type"
+                   id="type"
+                   hasNoPaddingTop>
+            <Radio id="live"
+                   name="type"
+                   label={_("Live")}
+                   isChecked={type === "live"}
+                   onChange={() => setType("live")} />
+            <Radio id="offline"
+                   name="type"
+                   label={_("Shut off")}
+                   isChecked={type === "offline"}
+                   onChange={() => {
+                       setType("offline");
+                       setStorage("shared");
+                   }} />
+        </FormGroup>
+    );
+};
+
+const UndefineSourceRow = ({ undefineSource, setUndefineSource }) => {
+    return (
+        <FormGroup label={_("Delete source VM")}
+                   fieldId="undefine-source"
+                   id="undefine-source"
+                   hasNoPaddingTop>
+            <Checkbox id="undefine-source"
+                      isChecked={undefineSource}
+                      label={_("Remove virtual machine from this host")}
```

Comment:
@garrett does this need an article? aka "Remove the virtual machine ..."

Reply:
**Quick summary**: UI text is often headline usage, where articles are dropped for simplicity.

This is sometimes pointed out in style guides, but often only implied.

---

UI text is often in classic* headline style of writing. Sometimes, this has included headline capitalization as well, but even in sentence capitalization, the text is still written like a headline.

_\* Of course wouldn't use the _clickbait_ headline style in a UI. :smirk: ("You wouldn't believe what this option does", "Six great tips for customizing virtual machines", "One sysadmin enabled shared storage and her story will astound you")_

In headline rules, you can — and should — drop additional words when it is clear what is being talked about. This conserves space and allows for more efficient scanning of the text.

Here's a good article about UI Copy (text in UIs) from Nielsen Norman Group (well-respected UI/UX designers who have written many books and articles):
https://www.nngroup.com/articles/ui-copy/

Relevant quote with emphasis:

> Keep command text short without sacrificing clarity. One-word, self-explanatory command text is easiest to scan, but obscure commands or those used infrequently will require more context. Use just enough text to accurately describe the command and include no more than 2–4 words.  Avoid overexplaining commands and **remove articles (“a”, “an”, and “the”) to speed up scannability, comprehension, and task completion**. While short commands are preferred, use enough text to explain the command sufficiently. 

For what it's worth, because we use headline style of text in UI, we often do drop punctuation, most commonly periods, at the end of a sentence... and often use phrases instead of sentences too.

For an example of a style guide that includes a suggestion on using this simplified type of text, please look at Google's Material design writing guidelines:
https://www.material.io/design/communication/writing.html

Most other places seem to imply this generally accepted UI text convention, including PatternFly. However, PatternFly does imply dropping unnecessary text and going for headline style text in many of its examples, such as [the writing error messages page](https://www.patternfly.org/v4/ux-writing/error-messages). (And [they do suggest simple text](https://www.patternfly.org/v4/ux-writing/ux-writing-best-practices#be-clear-and-concise)... however, they do not define exactly how it should be simple.)

Additionally, there's the concept of "[zero article](https://www.englishclub.com/grammar/determiners-zero-article.htm)", when articles such as "the" or "a" are often dropped from usage in English. It varies on UK vs. US English. However, headline rules and UI text go even beyond zero article or generic usage.

Reply:
FWIW: I looked this up, as I wanted to make 100% sure what I was suggesting was fine and also to share with others here.

You're more correct for English grammar. Normally, you would need an article here, but headline and UI text conventions "throw that out the window" (get rid of it).